### PR TITLE
ENG-5067: Added handling of correlation code

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java [hunk 1]
```diff
@@ -46,6 +46,8 @@ public class ResourceService implements IResourceService,
 
     private final EntLogger logger = EntLogFactory.getSanitizedLogger(this.getClass());
     
+    public static final String CORRELATION_CODE_PARAMETER = "cc";
```
Comment (entgigi) [inline cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java]: **[Major]** this is a duplicated constant pls use `ResourcesController.ID_NAME_OF_CORRELATION_CODE`

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java [hunk 2]
```diff
@@ -181,5 +185,9 @@ public PagedMetadata<ComponentUsageEntity> getComponentUsageDetails(String compo
         usageEntries.setBody(sublist);
         return usageEntries;
     }
+    
+    private String extractCorrelationCodeByCode(String code) {
+        return (code.startsWith(CORRELATION_CODE_PARAMETER + "=")) ? code.split("=")[1] : null;
```
Comment (entgigi) [inline cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java]: **[Major]** here I see 2 problems:
- you shouldn't propagate a `null` value, if you really don't have a default return `Optional.empty` and change the return type to `Optional<String>`
- you should reuse code from `ResourcesController.deleteAsset` if possible at least the logic so if not present the cc label you should use the value as code

**Review by entgigi — CHANGES_REQUESTED** [review]

**Review by mirkoPan — APPROVED** [review]
LGTM

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java (hunk 1 above)
Comment (eugeniosant) [reply cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java]: Done

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java (hunk 2 above)
Comment (eugeniosant) [reply cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java]: Done, now the method return an Optional and I'd tried to use the same logic of the controller

**Review by entgigi — APPROVED** [review]
LGTM

**Review by mirkoPan — APPROVED** [review]
LGTM

**Pull request merged**